specutils/tests/test_spectral_axis.py

```diff
@@ -0,0 +1,121 @@
+import astropy.units as u
+import numpy as np
+import pytest
+from astropy import time
+from astropy.constants import c
+from astropy.coordinates import (SkyCoord, EarthLocation, ICRS, GCRS, Galactic,
+                                 CartesianDifferential,
+                                 get_body_barycentric_posvel,
+                                 FK5, CartesianRepresentation)
+
+from ..spectra.spectral_coordinate import SpectralCoord
+from ..spectra.spectral_axis import SpectralAxis
+
+
+def get_greenwich_earthlocation():
+    """
+    A helper function to get an EarthLocation for greenwich (without trying to
+    do a download)
+    """
+    site_registry = EarthLocation._get_site_registry(force_builtin=True)
+    return site_registry.get('greenwich')
+
+def test_create_spectral_axis():
+
+    site = get_greenwich_earthlocation()
+    obstime = time.Time('2018-12-13 9:00')
+
+    observer_gcrs = site.get_gcrs(obstime)
+
+    wavelengths = np.linspace(500, 2500, 1001) * u.AA
+    spectral_axis = SpectralAxis(wavelengths, observer=observer_gcrs)
+
+    assert isinstance(spectral_axis, u.Quantity)
+    assert len(spectral_axis) == 1001
+    assert spectral_axis.bin_edges[0] == 499*u.AA
+
+def test_create_with_bin_edges():
+
+    wavelengths = np.linspace(500, 2500, 1001) * u.AA
+    spectral_axis = SpectralAxis(wavelengths, bin_specification="edges")
+
+    assert np.all(spectral_axis.bin_edges == wavelengths)
+    assert spectral_axis[0] == 501*u.AA
+
+    # Test irregular bin edges
+    wavelengths = np.array([500,510,550,560,590])*u.AA
+    spectral_axis = SpectralAxis(wavelengths, bin_specification="edges")
+
+    assert np.all(spectral_axis.bin_edges == wavelengths)
+    assert np.all(spectral_axis == [505., 530., 555., 575.]*u.AA)
+
+
+# GENERAL TESTS
```

Comment:
Am I right in guessing this is copied from `test_spectral_coord.py`?  If so, it might be more maintainable to use pytest's parametrized fixtures (https://docs.pytest.org/en/latest/fixture.html#parametrizing-fixtures) to have the same test function get run on both `SpectralCoord` and `SpectralAxis`?

Reply:
Yes, I did base most of this on the `test_spectral_coord.py` tests. Since I'd have to do a little reading about parametrizing fixtures, perhaps it would be better to merge this as-is and file an issue for consolidating some tests?